packages/esm-patient-registration-app/src/patient-registration/field/field.component.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { AddressField } from './address/address-field.component';
+import { PhoneEmailField } from './email/email-field.component';
+import { NameField } from './name/name-field.component';
+import { GenderField } from './gender/gender-field.component';
+import { IdField } from './id/id-field.component';
+import { DobField } from './dob/dob.component';
+import { reportError, useConfig } from '@openmrs/esm-framework';
+import { builtInFields, RegistrationConfig } from '../../config-schema';
+import { CustomField } from './custom-field.component';
+
+export interface FieldProps {
+  name: string;
+}
+
+export function Field({ name }: FieldProps) {
+  const config = useConfig() as RegistrationConfig;
+  if (
+    !(builtInFields as ReadonlyArray<string>).includes(name) &&
+    !config.fieldDefinitions.some((def) => def.id == name)
+  ) {
+    reportError(
+      `Invalid field name '${name}'. Valid options are '${config.fieldDefinitions
+        .map((def) => def.id)
+        .concat(builtInFields)
+        .join("', '")}'.`,
+    );
+    return null;
+  }
+
+  switch (name) {
+    case 'name':
+      return <NameField />;
+    case 'gender':
+      return <GenderField />;
+    case 'dob':
+      return <DobField />;
+    case 'address':
+      return <AddressField />;
+    case 'id':
+      return <IdField />;
+    case 'phone & email':
+      return <PhoneEmailField />;
```

Comment:
I suppose I forgot to mention this: one of the reasons I was trying to do away with the separate fields here is that they actually don't work as-is (mostly because they need to be submitted as attributes but aren't).

Reply:
Yeah it kind of looked like they actually do nothing? Which is frustrating